java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/converter/impl/BinaryAvaticaParameterConverter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.driver.jdbc.converter.impl;
+
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.calcite.avatica.AvaticaParameter;
+import org.apache.calcite.avatica.remote.TypedValue;
+
+/**
+ * AvaticaParameterConverter for Binary Arrow types.
+ */
+public class BinaryAvaticaParameterConverter extends BaseAvaticaParameterConverter {
+
+  public BinaryAvaticaParameterConverter(ArrowType.Binary type) {
+  }
+
+  @Override
+  public boolean bindParameter(FieldVector vector, TypedValue typedValue, int index) {
+    byte[] value = (byte[]) typedValue.toJdbc(null);
```

Comment:
Do we need to handle `value == null`? Or in general how does Avatica handle null bind parameters?

Reply:
Ohhh good catch. `Avatica` doesn't really do anything special for `null` values so it just returns `null`.

For example:

```
  public Object toJdbc(Calendar calendar) {
    return this.value == null ? null : serialToJdbc(this.type, this.componentType, this.value, calendar);
  }
```

I guess we need to call `setNull` if the value is null?

Reply:
Yeah - do you want to handle that here, or do you want to file a follow-up for now?

Reply:
Actually, let's just get this in before something else conflicts with it